modules/plugin/imagemosaic/src/test/java/org/geotools/gce/imagemosaic/HeterogenousCRSTest.java

```diff
@@ -958,4 +967,83 @@ public void testNativeEnvelope() throws Exception {
+        imReader.dispose();
+
+        // setup the datastore.properties wit the store name
+        final String TEST_STORE = "testStore";
+        File dsProperties = new File(testDirectory, "datastore.properties");
+        Properties properties = new Properties();
+        properties.put(Utils.Prop.STORE_NAME, "testStore");
+        try (FileOutputStream fos = new FileOutputStream(dsProperties)) {
+            properties.store(fos, null);
+        }
+
+        // wrap the shapefile store with one that records the Queries being used
+        URL shapefileURL = fileToUrl(new File(testDirectory, "hetero_s2.shp"));
+        ShapefileDataStore ds = new ShapefileDataStore(shapefileURL);
+        List<Query> queries = new ArrayList<>();
+        AtomicBoolean recordQueries = new AtomicBoolean(false);
+        DecoratingDataStore decoratingStore =
+                new DecoratingDataStore(ds) {
+                    @Override
+                    public SimpleFeatureSource getFeatureSource(String typeName)
+                            throws IOException {
+                        SimpleFeatureSource source = super.getFeatureSource(typeName);
+                        if (recordQueries.get()) {
+                            return new QueryCollectingFeatureSource(source, queries);
+                        }
+                        return source;
+                    }
+                };
+
+        // setup and pass the repository
+        DefaultRepository repository = new DefaultRepository();
+        repository.register(TEST_STORE, decoratingStore);
+
+        ImageMosaicReader repoReader =
+                new ImageMosaicReader(testDirectory, new Hints(Hints.REPOSITORY, repository));
+
+        // setup a grid geometry matching data, but a filter that won't let tiles be loaded
+        ParameterValue<GridGeometry2D> ggp = AbstractGridFormat.READ_GRIDGEOMETRY2D.createValue();
+        GridEnvelope2D range = new GridEnvelope2D(0, 0, 158, 103);
+        GridGeometry2D gg =
+                new GridGeometry2D(
+                        range,
+                        new ReferencedEnvelope(
+                                11.6834779,
+                                11.8616874,
+                                47.6380806,
+                                47.7542552,
+                                CRS.decode("EPSG:4236", true)));
+        ggp.setValue(gg);
+        ParameterValue<Filter> filter = ImageMosaicFormat.FILTER.createValue();
+        filter.setValue(CQL.toFilter("location = 'IAmNotThere.tif'"));
+
+        // record queries only during the read
+        recordQueries.set(true);
+        GridCoverage2D coverage = repoReader.read(new GeneralParameterValue[] {ggp, filter});
+        recordQueries.set(false);
+        if (coverage != null) coverage.dispose(true);
+        repoReader.dispose();
+        repository.dataStore(TEST_STORE).dispose();
+
+        // check the last query did not have the sorting, but the second to last, did not
```

Comment:
I think there is a small TYPO here. 
the second to last has sorting so probably the statement should end with "but the second to last did it".

Reply:
Improved explanation and improved assertions while I was at it.